src/ext/oblivc/obliv_bits.c

```diff
@@ -899,6 +861,41 @@ void yaoReleaseOt(ProtocolDesc* pd,int me)
   if(me==1) otSenderRelease(&ypd->sender);
   else otRecverRelease(&ypd->recver);
 }
+
+void splitProtocol(ProtocolDesc* pdout, ProtocolDesc * pdin) {
+  pdout->error= pdin->error;
+  pdout->partyCount = pdin->partyCount;
+  pdout->currentParty = pdin->currentParty;
+  pdout->feedOblivInputs = pdin->feedOblivInputs;
+  pdout->revealOblivBits = pdin->revealOblivBits;
+  pdout->setBitAnd = pdin->setBitAnd;
+  pdout->setBitOr = pdin->setBitOr;
+  pdout->setBitXor = pdin->setBitXor;
+  pdout->setBitNot = pdin->setBitNot;
+  pdout->flipBit = pdin->flipBit;
+  pdout->thisParty = pdin->thisParty;
+  pdout->trans = pdin->trans->split(pdin->trans);
+  pdout->copyextra = pdin->copyextra;
+  pdout->copyextra(pdout, pdin);
+}
```

Comment:
Yes, I think you're right. Functionally identical, but there's a significant advantage to preventing future mistakes. I'm pretty sure I caught all the places that updates were needed (there aren't too many), but if I take your suggestion, that'll ensure it.